### PR TITLE
update dex-server reconcile using apply

### PR DESCRIPTION
**Pull request by robinbobbitt**
https://github.com/open-cluster-management/backlog/issues/16514

### CODE REVIEW

**Review by cdoan1 — APPROVED** [review]
LGTM

File: deploy/dex-server/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+apiVersion: networking.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (robinbobbitt) [reply deploy/dex-server/ingress.yaml:1]: Done

**Review by cdoan1 — APPROVED** [review]
LGTM

**Pull request merged**